read.go

```diff
@@ -495,6 +493,24 @@ func ReadMetadata(inf IndexFile) ([]*Repository, *IndexMetadata, error) {
 	return rd.readMetadata(&toc)
 }
 
+// ReadMetadataPathAlive is like ReadMetadataPath except that it only returns
+// alive repositories. The order of repositories returned by
+// ReadMetadataPathAlive is not deterministic.
+func ReadMetadataPathAlive(p string) ([]*Repository, *IndexMetadata, error) {
+	repos, id, err := ReadMetadataPath(p)
+	if err != nil {
+		return nil, nil, err
+	}
+	cur := 0
+	for ix, repo := range repos {
+		if repo.Tombstone {
+			repos[cur], repos[ix] = repos[ix], repos[cur]
+			cur++
+		}
+	}
+	return repos[cur:], id, nil
```

Comment:
took me a few reads to validate this is correct. So you use the front of the slice to keep tombstoned repos. You could keep the "deterministic" ordering if you use the more classic go filtering pattern: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating

Reply:
deterministic is better :-)